doc/glossary.rst

```diff
@@ -0,0 +1,973 @@
+        independent variables.
+
+        Features in scikit-learn are generally expected to be numeric and
+        finite, even when they have semantically distinct domains and
+        distributions (categorical, ordinal, count-valued, real-valued,
+        interval). See also :term:`categorical feature` and :term:`missing
+        values`.
+
+        ``n_features`` indicates the number of features in a dataset.
+
+    fitting
+        Calling :term:`fit` (or :term:`fit_transform`, :term:`fit_predict`,
+        etc.) on an estimator.
+
+    fitted
+        The state of an estimator after :term:`fitting`.
+
+    function
+        TODO
+
+        Talk about where estimator fit or fit_transform functionality is present in a function.
+
+    joblib
+        A Python library (http://joblib.readthedocs.io) used in Scikit-learn to
+        facilite simple parallelism and caching.  Joblib is oriented towards
+        efficiently working with numpy arrays, such as through use of
+        :term:`memory mapping`.
+
+    leakage
+    data leakage
+        A problem in cross validation where generalization performance can be
+        over-estimated since knowledge of the test data was inadvertently
+        included in training a model.  This is a risk, for instance, when
+        applying a :term:`transformer` to the entirety of a dataset rather
+        than each training portion in a cross validation split.
+
+        We aim to provide interfaces (such as :mod:`pipeline` and
+        :mod:`model_selection`) that shield the user from data leakage.
+
+    memmapping
+    memory map
+    memory mapping
+        TODO
+
+    missing values
+        TODO
+
+    narrative docs
+    narrative documentation
+        An alias for :ref:`User Guide <user_guide>`, i.e. documentation written
+        in ``doc/modules/``.
+
+    ``n_features``
+        The number of :term:`features`.
+
+    ``n_outputs``
+        The number of :term:`outputs` in the :term:`target`.
+
+    ``n_samples``
+        The number of :term:`samples`.
+
+    ``n_targets``
+        Synonym for :term:`n_outputs`.
+
+    out-of-core
+        TODO
+
+    outputs
+        TODO?
+
+    parameter
+    parameters
+    param
+    params
+        We mostly use *parameter* to refer to the aspects of an estimator that
+        can be specified in its construction. For example, ``max_depth`` and
+        ``random_state`` are parameters of :class:`RandomForestClassifier`.
+        Parameters to an estimator's constructor are stored as attributes the
+        estimator, and conventionally start with an alphabetic character and
+        end with an alphanumeric character.  Each estimator's constructor
+        parameters are described in the estimator's docstring.
+
+        We do not use parameters in the statistical sense, where parameters are
+        values that specify a model and can be estimated from data. What we
+        call parameters might be what statisticians call hyperparameters to the
```

Comment:
I would move the first mention of the word hyper-parameter further up, but I like your explanation.